frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/ResetPasswordRequestHandler.java

```diff
@@ -0,0 +1,180 @@
+                                    LOGGER.info(
+                                            "Invalid session. Email {}",
+                                            resetPasswordRequest.getEmail());
+                                    return generateApiGatewayProxyErrorResponse(
+                                            400, ErrorResponse.ERROR_1000);
+                                }
+                                SessionState nextState =
+                                        stateMachine.transition(
+                                                session.get().getState(),
+                                                SYSTEM_HAS_SENT_RESET_PASSWORD_LINK);
+
+                                Optional<ErrorResponse> emailErrorResponse =
+                                        validationService.validateEmailAddress(
+                                                resetPasswordRequest.getEmail());
+                                if (emailErrorResponse.isPresent()) {
+                                    LOGGER.error(
+                                            "Encountered emailErrorResponse: {}",
+                                            emailErrorResponse.get());
+                                    return generateApiGatewayProxyErrorResponse(
+                                            400, emailErrorResponse.get());
+                                }
+                                return handleNotificationRequest(
+                                        resetPasswordRequest.getEmail(),
+                                        NotificationType.RESET_PASSWORD,
+                                        session.get(),
+                                        nextState);
+                            } catch (SdkClientException ex) {
+                                LOGGER.error("Error sending message to queue", ex);
+                                return generateApiGatewayProxyResponse(
+                                        500, "Error sending message to queue");
+                            } catch (JsonProcessingException e) {
+                                LOGGER.error("Error parsing request", e);
+                                return generateApiGatewayProxyErrorResponse(400, ERROR_1001);
+                            } catch (StateMachine.InvalidStateTransitionException e) {
+                                LOGGER.error("Invalid transition in user journey", e);
+                                return generateApiGatewayProxyErrorResponse(400, ERROR_1017);
+                            }
+                        });
+    }
+
+    private APIGatewayProxyResponseEvent handleNotificationRequest(
+            String email,
+            NotificationType notificationType,
+            Session session,
+            SessionState nextState)
+            throws JsonProcessingException {
+
+        String subjectId = authenticationService.getSubjectFromEmail(email).getValue();
```

Comment:
Do we expect this to always find the user or is there a chance that a user might not exist with that email?

Reply:
They have to have gone through checkuserexists to get here so in theory the user should exist.  It's possible if someone is running two flows at the same time and deletes the user.  Could make it return an error?

Reply:
This is a very rare edge case as the user would have to delete their account between the time they enter their email and clicking the check password link therefore it we can let it blow up if the user isn't there. No action needed